File: 02-java-exercises/3-factorial-java/src/Factorial.java
public class Factorial {

    public static void main(String[] args) {

        int number = 5;

        System.out.println(number + "! = " + factorialIterative(number));
        System.out.println(number + "! = " + factorialRecursive(number));
    }

    /* FACTORIAL (iterative) */
    public static int factorialIterative(int number) {
        int result = number;
        for (int i = number; i > 1; i--) {
            result = result * (i - 1);
        }
        return result;
    }

    /* FACTORIAL (recursive) */
    public static int factorialRecursive(int number) {
        if (number == 1) {
            return number;
        } else {
            return number * factorialRecursive(number - 1);
        }
    }
}

File: 3-factorial/factorial.js
/* FACTORIAL */

var numberToCalculate = 5;

console.log("// Factorial (iterative) //");

function factorialIt(number) {
  var result = number;
  for (var i = number; i>1; i--) {
    result = result * (i-1);
  }
  return result;
}
console.log(numberToCalculate + "! = " + factorialIt(numberToCalculate));


console.log("\n");


console.log("// Factorial (recursive) //");

function factorial(number) {
  if (number === 1) {
    return number;
  } else {
    return number * factorial(number - 1);
  }
}
console.log(numberToCalculate + "! = " + factorial(numberToCalculate));
